src/error.rs

```diff
@@ -1,35 +1,136 @@
 // Copyright 2016, The Gtk-rs Project Developers.
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+use std::fmt::{self, Debug};
 
-use enums::Status;
 use std::io;
 use thiserror::Error;
 
 #[derive(Error, Debug)]
-pub enum BorrowError {
-    #[error("Failed to borrow with Cairo error:{0}")]
-    Cairo(Status),
-    #[error("Can't get exclusive access")]
-    NonExclusive,
+pub enum Error {
+    NoMemory,
+    InvalidRestore,
+    InvalidPopGroup,
+    NoCurrentPoint,
+    InvalidMatrix,
+    InvalidStatus,
+    NullPointer,
+    InvalidString,
+    InvalidPathData,
+    ReadError,
+    WriteError,
+    SurfaceFinished,
+    SurfaceTypeMismatch,
+    PatternTypeMismatch,
+    InvalidContent,
+    InvalidFormat,
+    InvalidVisual,
+    FileNotFound,
+    InvalidDash,
+    InvalidDscComment,
+    InvalidIndex,
+    ClipNotRepresentable,
+    TempFileError,
+    InvalidStride,
+    FontTypeMismatch,
+    UserFontImmutable,
+    UserFontError,
+    NegativeCount,
+    InvalidClusters,
+    InvalidSlant,
+    InvalidWeight,
+    InvalidSize,
+    UserFontNotImplemented,
+    DeviceTypeMismatch,
+    DeviceError,
+    InvalidMeshConstruction,
+    DeviceFinished,
+    JBig2GlobalMissing,
+    PngError,
+    FreetypeError,
+    Win32GdiError,
+    LastStatus,
+    Unknown,
 }
 
-impl From<Status> for BorrowError {
-    fn from(status: Status) -> Self {
-        BorrowError::Cairo(status)
+impl fmt::Display for Error {
```

Comment:
You can also just derive `Display` via `thiserror` (and we should give them better strings than what we have below...)